Sources/_Datadog_TestRunner/DDTestRunner.swift

```diff
@@ -0,0 +1,138 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+@_implementationOnly import XCTest
+
+internal class DDTestRunner: NSObject, XCTestObservation {
+    static var instance: DDTestRunner?
+
+    let testNameRegex = try? NSRegularExpression(pattern: "([\\w]+) ([\\w]+)", options: .caseInsensitive)
+    let supportsSkipping = NSClassFromString("XCTSkippedTestContext") != nil
+    var currentBundleName = ""
+    var activeTestSpan: DDSpan?
+
+    override init() {
+        super.init()
+        XCTestObservationCenter.shared.addTestObserver(self)
+        startTracer()
+    }
+
+    func startTracer() {
+        var clientToken = ""
+        if let envToken = ProcessInfo.processInfo.environment["DATADOG_CLIENT_TOKEN"] {
+            clientToken = envToken.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+        } else if let bundleToken = Bundle.main.infoDictionary?["DatadogClientToken"] as? String {
+            clientToken = bundleToken.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+        }
```

Comment:
Some `ProcessInfo` ENVs are abstracted by the `CIEnvironmentValues` and some are not. How about moving all ENVs consideration to the `CIEnvironmentValues` and just keeping the `env` on a single class property:
```swift
internal class DDTestRunner: NSObject, XCTestObservation {
   private let env = CIEnvironmentValues()

   // ...
   // env.datadogClientToken, ...
}
```

This way we could also test `DATADOG_CLIENT_TOKEN` and `DatadogClientToken` retrieval in `CIEnvironmentValuesTests`. Right now it's not tested.

Reply:
Modified to store also Datadog environment values (DATADOG_CLIENT_TOKEN, DD_SERVICE and DD_ENV), and not only CI values. 
Renamed to DDEnvironmentValues. 
Added test for these new values